src/compas_fab/robots/robot.py

```diff
@@ -800,19 +800,21 @@ def attach_tool(self, tool, group=None, touch_links=None):
 
         Examples
         --------
-        >>> mesh = Mesh.from_stl(compas_fab.get('planning_scene/cone.stl'))
+        >>> mesh = Mesh.from_stl(compas.get('cone.stl'))
         >>> frame = Frame([0.14, 0, 0], [0, 1, 0], [0, 0, 1])
         >>> tool = Tool(mesh, frame)
         >>> robot.attach_tool(tool)
         """
-        group = group or self.main_group_name
-        ee_link_name = self.get_end_effector_link_name(group)
-        touch_links = touch_links or [ee_link_name]
-        tool.attached_collision_mesh = AttachedCollisionMesh(tool.collision_mesh, ee_link_name, touch_links)
+        if not tool.link_name:
+            group = group or self.main_group_name
+            tool.link_name = self.get_end_effector_link_name(group)
+        tool.update_touch_links(touch_links)
         self.attached_tool = tool
         if self.artist:
-            self.update(self.zero_configuration(), group=group, visual=True, collision=True)  # TODO: this is not so ideal! should be called from within artist
-            self.artist.attach_tool(tool)
+            zero_config = self.zero_configuration()
+            joint_state = dict(zip(zero_config.joint_names, zero_config.values))
+            self.update(joint_state, visual=True, collision=True)  # TODO: this is not so ideal! should be called from within artist
+            self.artist.attach_tool_model(tool.tool_model)
```

Comment:
Is this still necessary with the (perhaps unsafe) change I made to `attached_tool_model`?

Reply:
I decided that it isn't necessary.